runtime/lua/vim/lsp.lua

```diff
@@ -25,13 +25,40 @@ local lsp = {
   -- format_rpc_error = lsp_rpc.format_rpc_error;
 }
 
+-- maps request name to the required resolved_capability in the client.
+local req_to_capability = {
```

Comment:
(unrelated to this PR) why do we have different key names that require this mapping? why don't we use the protocol-defined names?

Reply:
not sure if I follow the comment, sorry about that. Do you mean introducing some shared variables for the method names or something like that? Or if the protocol itself already declares that dependency-mapping? Or something else? x)

Reply:
These are the protocol defined names, I think it's fine.